source/postprocess/plateness_statistics.cc

```diff
@@ -0,0 +1,196 @@
+
+                this->get_material_model().fill_additional_material_model_inputs(in,
+                                                                                 this->get_solution(),
+                                                                                 fe_face_values,
+                                                                                 this->introspection());
+
+                this->get_material_model().evaluate(in, out);
+
+                for (unsigned int q = 0; q<fe_face_values.n_quadrature_points; ++q)
+                  {
+                    // Calculate the square root of the second moment invariant for the deviatoric strain rate tensor.
+                    //edot_ii = std::max(std::sqrt(std::fabs(second_invariant(deviator(in.strain_rate[q])))),
+                    //          min_strain_rate);
+
+                    const double local_second_invariant_of_strain_rate = (std::max(std::sqrt(std::fabs(second_invariant(deviator(in.strain_rate[q])))),min_strain_rate) * fe_face_values.JxW(q));
+                    const double local_surface_area = fe_face_values.JxW(q);
+
+                    // second invariant of strain rate over the whole surface
+                    local_second_invariant_of_strain_rate_integral += local_second_invariant_of_strain_rate;
+                    local_surface_area_integral += local_surface_area;
+                  }
+              }
+          }
+
+      double total_second_invariant_of_strain_rate (local_second_invariant_of_strain_rate_integral.size());
+      Utilities::MPI::sum (local_second_invariant_of_strain_rate_integral, this->get_mpi_communicator(), total_second_invariant_of_strain_rate);
+
+      double total_surface_area (local_surface_area_integral.size());
+      Utilities::MPI::sum (local_surface_area_integral, this->get_mpi_communicator(), total_surface_area);
+
+
+      //Merge so column 1 is strain rate invariants and column 2 are the corresponding areas
+      for (int i = 0; i < local_second_invariant_of_strain_rate_integral.size(); ++i)
+        {
+          local_second_invariant_of_strain_rate_and_area[i][0] = local_second_invariant_of_strain_rate_and_area[i];
+          local_second_invariant_of_strain_rate_and_area[i][1] = local_surface_area_integral[i];
+        }
+
+      //sort local second invariants of strain rate in size order
+      sorted_local_second_invariant_of_strain_rate = sort(local_second_invariant_of_strain_rate[:][0].begin(), local_second_invariant_of_strain_rate[:][0].end(), greater<int>());
+
+      //find the fraction of the surface area where 80% of the deformation occurs
+      //double f_80 = 0.0;
+      double cumulative_second_invariant_of_strain_rate = 0.0;
+      double cumulative_surface_area = 0.0;
+      int row_number = 0;
+      do
+        {
+          cumulative_second_invariant_of_strain_rate = cumulative_second_invariant_of_strain_rate + local_second_invariant_of_strain_rate_and_area[row_number][0];
```

Comment:
This type of loop is somewhat dangerous. If the while condition is not well defined you may end up accessing memory beyond the end of local_second_invariant_of_strain_rate_and_area. Better add this, which will cause ASPECT to crash with an error message if we ever not meet the while condition.
```suggestion
      Assert(row_number < local_second_invariant_of_strain_rate_and_area.size(), ExcInternalError());
```

Reply:
Ok thanks. I assume you don't mean to remove line 144 also "cumulative_second_invariant_of_strain_rate = cumulative_second_invariant_of_strain_rate + local_second_invariant_of_strain_rate_and_area[row_number][0];". We still need that.